misk/src/main/kotlin/misk/web/WebModule.kt

```diff
@@ -42,13 +47,35 @@ class WebModule : KAbstractModule() {
         // Create an empty set binder of interceptor factories that can be added to by users.
         newSetBinder<Interceptor.Factory>()
 
-        // Register built-in interceptors
+        // Register built-in interceptors. Interceptors run in the order in which they are
+        // installed, and the order of these interceptors is critical.
+
+        // Handle all unexpected errors that occur during dispatch
         binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<InternalErrorInterceptorFactory>()
+
+        // Optionally log request and response details
         binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<RequestLoggingInterceptor.Factory>()
+
+        // Collect metrics on the status of results and response times of requests
         binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<MetricsInterceptor.Factory>()
+
+        // Convert application level exceptions into their appropriate response format
+        binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<ExceptionMappingInterceptor.Factory>()
+
+        // Log exceptions that come from the application at the appropriate level
+        binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<ExceptionLoggingInterceptor.Factory>()
+
+        // Convert typed responses into a ResponseBody that can marshal the response according to
+        // the client's requested content-typ
         binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<MarshallerInterceptor.Factory>()
+
+        // Wrap "raw" responses with a Response object
         binder().addMultibinderBindingWithAnnotation<Interceptor.Factory, MiskDefault>().to<BoxResponseInterceptorFactory>()
 
+        // Register build-in exception mappers
+        install(ExceptionMapperModule.create<ActionExceptionMapper>())
+        install(ExceptionMapperModule.create<IllegalArgumentExceptionMapper>())
```

Comment:
I’m not sold on this one. I think if the application layer bubbles up an IllegalArgumentException, it’s just as likely to be the server’s fault as the caller’s